pyproject.toml

```diff
@@ -111,6 +111,11 @@ allow-prereleases = false
 python = "^3.6"
 markers = "platform_python_implementation == 'CPython'"
 
+
+[tool.poetry.group.dev.dependencies]
+mypy = "^0.991"
+types-tqdm = "^4.64.7.9"
+
```

Comment:
Currently in `[tool.poetry.dependencies]`, numpy is pinned to `numpy = "^1.19"` (not sure how poetry handles this, but `numpy.typing.NDArray` is new in numpy 1.21, so we should probably bump it.

Reply:
That's not a big problem, the type checking with mypy is independent because you can choose python version independent of your interpreter. But i also didn't get the types for numpy to work at the current stage of repo.

Reply:
What I meant is you are using `npt.NDArray` in `manim.typing` and type aliases defined in this module are used in a bunch of places in the code, without a `if TYPE_CHECKING` check, so if people are running a version of `numpy<1.21`, it would fail at runtime